common/djangoapps/third_party_auth/models.py

```diff
@@ -75,6 +77,130 @@ def __str__(self):
             self.provider_name
         )
 
+class UserDataSharingConsentAudit(TimeStampedModel):
+    """
+    Object that exists to store the canonical state of whether a particular
+    user has given consent for their course data to be shared with a particular
+    single-sign-on identity provider
+    """
+    class Meta(object):
+        app_label = 'third_party_auth'
+        verbose_name = "Data Sharing Consent Audit State"
+        verbose_name_plural = "Data Sharing Consent Audit States"
+
+    NOT_SET = 'NS'
+    ENABLED = 'EN'
+    DISABLED = 'DS'
```

Comment:
@haikuginger nit: might be good to use more verbose codes, like `not_set`, `enabled` and `disabled`.